feg/gateway/sbi/sbi_test.go

```diff
@@ -0,0 +1,158 @@
+	n7 "magma/feg/gateway/sbi/specs/TS29512NpcfSMPolicyControlServer"
+	sbi "magma/feg/gateway/sbi/specs/TS29571CommonData"
+)
+
+type MockPcf struct {
+	ListenAddr string
+	Policies   map[string]n7.SmPolicyControl
+}
+
+func NewMockPcf() *MockPcf {
+	return &MockPcf{
+		Policies: make(map[string]n7.SmPolicyControl),
+	}
+}
+
+func (pcf *MockPcf) PostSmPolicies(ctx echo.Context) error {
+	log.Println("PostSmPolicies")
+	var newPolicy n7.SmPolicyContextData
+	err := ctx.Bind(&newPolicy)
+	if err != nil {
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	var policy n7.SmPolicyControl
+	policy.Context = newPolicy
+
+	pcf.Policies["1234"] = policy
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+// GetSmPoliciesSmPolicyId handles GET /sm-policies/{smPolicyId}
+func (pcf *MockPcf) GetSmPoliciesSmPolicyId(ctx echo.Context, smPolicyId string) error {
+	policy, found := pcf.Policies[smPolicyId]
+	if !found {
+		return ctx.NoContent(http.StatusNotFound)
+	}
+	return ctx.JSON(http.StatusOK, policy)
+}
+
+// PostSmPoliciesSmPolicyIdDelete handles POST /sm-policies/{smPolicyId}/delete
+func (pcf *MockPcf) PostSmPoliciesSmPolicyIdDelete(ctx echo.Context, smPolicyId string) error {
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+// PostSmPoliciesSmPolicyIdUpdate handles POST /sm-policies/{smPolicyId}/update
+func (pcf *MockPcf) PostSmPoliciesSmPolicyIdUpdate(ctx echo.Context, smPolicyId string) error {
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+func TestCreateSMPolicy(t *testing.T) {
+	// Start the server and wait till the server is started
+	mockPcf := startServer(t)
+
+	// Create new N7 client object
+	client, err := n7_client.NewClientWithResponses(fmt.Sprintf("http://%s", mockPcf.ListenAddr))
+
+	assert.NoError(t, err, "Client creation failed")
+
+	// Create a new SM Policy
+	ipAddr := sbi.Ipv4Addr("10.1.2.3")
+	imsi := sbi.Supi("12345678901234")
+	body := n7_client.PostSmPoliciesJSONRequestBody{
+		Ipv4Address:  &ipAddr,
+		PduSessionId: 10,
+		Supi:         imsi,
+	}
+
+	resp, err := client.PostSmPolicies(context.Background(), body)
+	assert.NoError(t, err, "Failed to create SM Policies")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	// Get SM Policy given the policyId
+	policyResp, err := client.GetSmPoliciesSmPolicyIdWithResponse(context.Background(), "1234")
+	assert.NoError(t, err, "Failed to get SM Policies")
+	assert.Equal(t, http.StatusOK, policyResp.StatusCode())
+	assert.Equal(t, &ipAddr, policyResp.JSON200.Context.Ipv4Address)
+	assert.Equal(t, imsi, policyResp.JSON200.Context.Supi)
+}
+
+func startServer(t *testing.T) *MockPcf {
+	errChan := make(chan error)
+	eserver := echo.New()
+
+	mockPcf := NewMockPcf()
+	go func() {
+		n7.RegisterHandlers(eserver, mockPcf)
+
+		err := eserver.Start("localhost:0")
+		if err != nil {
+			errChan <- err
+		}
+	}()
+
+	addr, err := waitForServerStart(eserver, errChan)
+
+	assert.NoError(t, err)
+
+	mockPcf.ListenAddr = addr
+
+	return mockPcf
```

Comment:
try not to use so many blank lines 

This may look better
```
	mockPcf.ListenAddr, err := waitForServerStart(eserver, errChan)
	assert.NoError(t, err)
	return mockPcf
```

Reply:
This results in the following error.
`sbi/sbi_test.go:127:2: expected identifier on left side of :=`

The variable err was not defined before.

Reply:
remove the ':'


```
mockPcf.ListenAddr, err = waitForServerStart(eserver, errChan) 
assert.NoError(t, err) 
return mockPcf
``